base/compiler/inferencestate.jl

```diff
@@ -1116,3 +1121,74 @@ function get_max_methods_for_module(mod::Module)
     max_methods < 0 && return nothing
     return max_methods
 end
+
+"""
+    Future{T}
+
+Delayed return value for a value of type `T`, similar to RefValue{T}, but
+explicitly represents completed as a `Bool` rather than as `isdefined`.
+Set once with `f[] = v` and accessed with `f[]` afterwards.
+
+Can also be constructed with the `completed` flag value and a closure to
+produce `x`, as well as the additional arguments to avoid always capturing the
+same couple of values.
```

Comment:
It looks like this feature isn't implemented? Or we could refactor the constructor that takes `interp` and `sv` below and split it into the pure core functionality of `Future` and a separate function for `AbsIntState` that uses it.

Reply:
It is implemented. The additional arguments are `interp` and `sv`. Looks like these could use splatting for them, but didn't seem like much point, and that it could be more confusing.